src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.exception.DukeExceptionType;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Storage {
+    private static final String TASK_SEPARATOR = "~~";
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath.replace('/', File.separatorChar);
+    }
+
+    public String[] load() throws DukeException {
+        File f = new File(filePath);
+        String[] tasks;
+        try {
+             tasks = getTasks(f);
+        } catch (FileNotFoundException e) {
+            throw new DukeException(DukeExceptionType.ERROR_LOADING_FILE);
+        }
+        return tasks;
+    }
+
+    private String[] getTasks(File f) throws FileNotFoundException {
+        StringBuilder data = new StringBuilder();
+        int taskCount;
+        taskCount = readTasksAndGetTaskCount(data, f);
+        String[] tasks = data.toString().split(TASK_SEPARATOR, taskCount);
+        return tasks;
+    }
+
+    private int readTasksAndGetTaskCount(StringBuilder data, File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f);
+        int taskCount = 0;
+        while (s.hasNextLine()) {
+            String line = s.nextLine();
+            data.append(line);
+            taskCount++;
+            if (s.hasNextLine()) {
+                data.append(TASK_SEPARATOR);
+            }
+        }
+        return taskCount;
+    }
```

Comment:
There are two ways you can improve the code. 
- You can split the functionality to two methods. Read data in one and compute the count in the other
- OR use a collection and get the size for task count.